src/docs/sphinx/dev_guide/tensor_dual.rst

```diff
@@ -0,0 +1,405 @@
+.. _header-n0:
+
+============
+Tensor Class
+============
+
+``tensor`` is a class template for doing arithmetic on small,
+statically-sized vectors, matrices and tensors. To create one, specify
+the underlying type in the first template argument, followed by a list
+of integers for the shape. For example, ``tensor<float,3,2,4>`` is
+conceptually similar to the type ``float[3][2][4]``.
+
+Here are some examples and features:
+
+-  ``tensor`` has value semantics (in contrast to C++ multidimensional
+   arrays):
+
+   .. code-block:: c++
+
+      {
```

Comment:
Can you break this into two different examples and/or put some more comments for what each scope is doing?  

Also either link to what value semantics means or add a sentence explaining it.  This will be read by multiple audiences and they may not understand what you mean by that.